### ISSUE TITLE
Batch distribution using deeplake dataset.

### ISSUE THREAD

**Issue by Andrea-ghub** [selimfirat/greenedge-energy-aware-classification#1]
Here you can see an image of the label distribution in batches of 256 images. The batches are done with the following snippet:
`
train_data = deeplake.load("hub://activeloop/tiny-imagenet-train")
`
`dataloader = train_data.pytorch(
    num_workers=0,
    tensors = ['images', 'labels'],
    transform = tiny_image_net_transform(tform),
    batch_size=256,
    shuffle=True
)`

This is the label distribution for the first 40 batches:

![e5df4d26-1433-455d-ac02-e62a86729131](https://github.com/selimfirat/greenedge-energy-aware-classification/assets/59959945/8a70c61a-721b-4c99-a119-d55ec7165106)

You can notice that the distribution is not uniform over the 200 labels but every batch contains just around 17 labels per batch. 

You can find the notebook used to obtain the results at this link (@https://colab.research.google.com/drive/118tVGt7Fqwk5FAqAKBnI_-ONc8LlAUS6?usp=sharing), hope everything is correct.

Andrea and Luca :)

**Comment by selimfirat** [comment]
Hi Andrea,

This looks interesting. Thanks for pointing out.

I think deeplake uniformizes the points based on a fixed cache, so it cannot be fully randomized. I suggest to increase memory_cache_size hyperparameter or change access_method when calling .load().

Could you please try with increased cache size and access_method="download" or access_method="local" and let us know?

https://docs.deeplake.ai/en/latest/deeplake.html

In the meantime, please feel free to implement your own loader after loading whole dataset to disk or memory.
